File: src/dataset/go/etl_Exercism_go_drapala/openAI/ChatGPT4_etl_v3.go
package etl

import "strings"

func Transform(in map[int][]string) map[string]int {
	result := make(map[string]int, len(in)*len(in)) // Preallocate with an estimated size

	for key, value := range in {
		for i := 0; i < len(value); i++ {
			result[strings.ToLower(value[i])] = key
		}
	}
	return result
}

File: src/dataset/cpp/etl_exercism-johnngugi/openAI/ChatGPT4_etl_v3.cpp
#include "etl.h"
#include <map>
#include <vector>
#include <cctype>
using namespace std;

map<char, int> etl::transform(const map<int, vector<char>>& a)
{
    map<char, int> store;
    for (const auto& [key, chars] : a)
    {
        for (char c : chars)
        {
            store[tolower(c)] = key;
        }
    }
    return store;
}